duckduckgo/macos-browser
# Implement Phishing Detection Integration

Task/Issue URL: https://app.asana.com/0/1204023833050360/1207334681074772/f
Tech Design URL: https://app.asana.com/0/481882893211075/1207220724600204/f
CC: @SabrinaTardio 

**Description**:
This PR introduces:
1. PhishingDetectionManager.swift: a wrapper singleton class around PhishingService.swift and PhishingDataActivities.swift (in BSK) - this class should kick-off NSBackgroundActivity scheduled tasks for updating the phishing datasets. This happens in `applicationDidFinishLaunching`.
2. PhishingErrorPageHTMLTemplate: implements the userText strings for the HTML error page (with localizable strings)
3. PhishingErrorPageUserScript: the message handlers for the error page user script
4. PhishingErrorPageTabExtension: handles the navigation interception, URL check, and leaveSite/visitSite functionality

**Steps to test this PR**:
Sorry, there are lots of prerequisites here, so bear with me:
1. You'll need to use [my C-S-S branch](https://github.com/duckduckgo/content-scope-scripts/tree/tespach/phishing-error-page) to get the actual error page HTML.
2. Remember to `npm build` the C-S-S repo and update the branch in Xcode
3. You'll also need [my BSK branch](https://github.com/duckduckgo/BrowserServicesKit/tree/tespach/phishing-protection-client) to get PhishingDetectionService etc.
4. Finally, you'll need to mock the API (for now.. BE should be deployed imminently!) [Use this script.](https://dub.duckduckgo.com/gist/tespach/d1e90128083d30077edc520fa641c3ef)

Now to test:
1. Browse as normal, ensure everything loads and runs as expected
2. Now browse to https://privacy-test-pages.site - ensure the error page is not shown here.
3. Under **Security** click "An example phishing page"
4. Ensure the error page shows up
5. Now click "Advanced>Visit Site" - ensure the actual page loads
6. Refresh page, ensure the page still loads
7. Navigate backwards, ensure the Privacy Test Pages home page is shown
8. Play around with refresh/back/forward here to ensure everything works as expected
9. Open a new tab, visit http://bad.third-party.site/security/badware/phishing.html, ensure error page is thrown
10. Hit "Leave Site"
11. Ensure you're taken to the new tab page



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
